docs/optimistic_locking.rst

```diff
@@ -0,0 +1,175 @@
+==================
+Optimistic Locking
+==================
+
+Optimistic Locking is a strategy for ensuring that your database writes are not overwritten by the writes of others.
+With optimistic locking, each item has an attribute that acts as a version number. If you retrieve an item from a
+table, the application records the version number of that item. You can update the item, but only if the version number
+on the server side has not changed. If there is a version mismatch, it means that someone else has modified the item
+before you did. The update attempt fails, because you have a stale version of the item. If this happens, you simply
+try again by retrieving the item and then trying to update it. Optimistic locking prevents you from accidentally
+overwriting changes that were made by others. It also prevents others from accidentally overwriting your changes.
+
+Note
```

Comment:
nit: can you use the `warning` directive here instead? i.e

```
.. warning::
   warning goes here
```

See docs/backup_restore.rst for an example

Reply:
Switched to a warning